manageiq-operator/pkg/helpers/miq-components/httpd.go

```diff
@@ -278,197 +303,195 @@ func initializeHttpdContainer(spec *miqv1alpha1.ManageIQSpec, privileged bool, c
 	return nil
 }
 
-func NewHttpdDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, error) {
+func HttpdDeployment(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*appsv1.Deployment, controllerutil.MutateFn, error) {
 	privileged, err := PrivilegedHttpd(cr.Spec.HttpdAuthenticationType)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	container := corev1.Container{}
 	err = initializeHttpdContainer(&cr.Spec, privileged, &container)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	deploymentLabels := map[string]string{
-		"app": cr.Spec.AppName,
-	}
-	podLabels := map[string]string{
-		"name": "httpd",
 		"app":  cr.Spec.AppName,
+		"name": "httpd",
 	}
-	var repNum int32 = 1
 
 	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "httpd",
 			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    deploymentLabels,
 		},
 		Spec: appsv1.DeploymentSpec{
-			Replicas: &repNum,
 			Selector: &metav1.LabelSelector{
-				MatchLabels: podLabels,
+				MatchLabels: deploymentLabels,
 			},
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
+					Labels: deploymentLabels,
 					Name:   "httpd",
-					Labels: podLabels,
-				},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{container},
-					Volumes: []corev1.Volume{
-						corev1.Volume{
-							Name: "httpd-config",
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{Name: "httpd-configs"},
-								},
-							},
-						},
-					},
 				},
+				Spec: corev1.PodSpec{},
 			},
 		},
 	}
 
-	configureHttpdAuth(&cr.Spec, &deployment.Spec.Template.Spec)
-
 	// Only assign the service account if we need additional privileges
 	if privileged {
 		deployment.Spec.Template.Spec.ServiceAccountName = cr.Spec.AppName + "-httpd"
```

Comment:
This should be in the mutate function, right?

Reply:
Done